### PR TITLE
Replace lazy_static with LazyLock

### LINKED ISSUE #5641: Replace `lazy_static` with `OnceLock`
We should be able to replace all `lazy_static` usages with `OnceLock` from the Rust standard library. This will bring us less dependencies. [The lazy_static README](https://github.com/rust-lang-nursery/lazy-static.rs?tab=readme-ov-file#standard-library) gives an example on how to do this.

### PR DESCRIPTION
**Pull request by programskillforverification**
# Description

close #5641

### CODE REVIEW

File: polkadot/node/network/dispute-distribution/src/tests/mock.rs [hunk 1]
```diff
@@ -60,64 +59,59 @@ pub const ALICE_INDEX: ValidatorIndex = ValidatorIndex(1);
 pub const BOB_INDEX: ValidatorIndex = ValidatorIndex(2);
 pub const CHARLIE_INDEX: ValidatorIndex = ValidatorIndex(3);
 
-lazy_static! {
-
 /// Mocked `AuthorityDiscovery` service.
-pub static ref MOCK_AUTHORITY_DISCOVERY: MockAuthorityDiscovery = MockAuthorityDiscovery::new();
+pub static MOCK_AUTHORITY_DISCOVERY: OnceLock<MockAuthorityDiscovery> = OnceLock::new();
```
Comment (bkchr) [inline polkadot/node/network/dispute-distribution/src/tests/mock.rs:63]: We can also use a LazyLock here?

**Review by bkchr — APPROVED** [review]
Ty!

I think we can remove the one and only `OnceLock` and replace this with a `LazyLock` as well :D

File: polkadot/node/network/dispute-distribution/src/tests/mock.rs (hunk 1 above)
Comment (programskillforverification) [reply polkadot/node/network/dispute-distribution/src/tests/mock.rs:63]: > We can also use a LazyLock here?

I forgot the one.

File: polkadot/node/zombienet-backchannel/Cargo.toml [hunk 2]
```diff
@@ -12,11 +12,10 @@ license.workspace = true
 workspace = true
 
 [dependencies]
-tokio = { features = ["macros", "net", "rt-multi-thread", "sync"], workspace = true }
+tokio = { features = ["macros", "net", "rt-multi-thread", "sync",], workspace = true }
```
Comment (michalkucharczyk) [inline polkadot/node/zombienet-backchannel/Cargo.toml:15]:
```suggestion
tokio = { features = ["macros", "net", "rt-multi-thread", "sync"], workspace = true }
```

**Review by michalkucharczyk — APPROVED** [review]

File: polkadot/node/zombienet-backchannel/Cargo.toml (hunk 2 above)
Comment (michalkucharczyk) [reply polkadot/node/zombienet-backchannel/Cargo.toml:15]: nit: `taplo` util shall be used to format `Cargo.toml` files.

File: cumulus/pallets/parachain-system/src/tests.rs [hunk 3]
```diff
@@ -754,31 +755,27 @@ fn message_queue_chain() {
...
 #[cfg(not(feature = "runtime-benchmarks"))]
 fn receive_dmp() {
-	lazy_static::lazy_static! {
-		static ref MSG: InboundDownwardMessage = InboundDownwardMessage {
-			sent_at: 1,
-			msg: b"down".to_vec(),
-		};
-	}
+	static MSG: OnceLock<InboundDownwardMessage> = OnceLock::new();
+	MSG.get_or_init(|| InboundDownwardMessage { sent_at: 1, msg: b"down".to_vec() });
```
Comment (bkchr) [inline cumulus/pallets/parachain-system/src/tests.rs:759]: No need at all that this is `OnceLock`.

Comment (bkchr) [reply cumulus/pallets/parachain-system/src/tests.rs:759]: @programskillforverification after this is done, we can merge it.

File: prdoc/pr_5716.prdoc [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  - name: frame-benchmarking-cli
+    bump: Major
+  - name: sc-offchain
+    bump: Patch
+  - name: polkadot-dispute-distribution
+    bump: Patch
+  - name: polkadot-gossip-support
+    bump: Patch
+  - name: xcm-emulator
+    bump: Patch
```
Comment (bkchr) [inline prdoc/pr_5716.prdoc:37]:
```suggestion
  - name: sp-core
    bump: patch
  - name: sp-panic-handler
    bump: patch
  - name: sp-trie
    bump: patch
  - name: sc-utils
    bump: major
  - name: cumulus-pallet-parachain-system
    bump: patch
  - name: sp-consensus-beefy
    bump: patch
  - name: polkadot-node-primitives
    bump: patch
  - name: polkadot-node-jaeger
    bump: patch
  - name: frame-benchmarking-cli
    bump: major
  - name: sc-offchain
    bump: patch
  - name: polkadot-dispute-distribution
    bump: patch
  - name: polkadot-gossip-support
    bump: patch
  - name: xcm-emulator
    bump: patch  
```

**Pull request merged**